src/reader.ts

```diff
@@ -0,0 +1,17 @@
+export interface Reader<T, U> {
+  map<K>(fn: (value: U) => K): Reader<T, K>,
+  flatMap<K>(fn: (value: U) => Reader<T, K>): Reader<T, K>,
+  run:(config: T) => U
+}
+
+export function Reader<T, U>(run: (ctx: T) => U): Reader<T, U> {
+  return {
+    map: function(fn) {
```

Comment:
As per the other implementations, please use the arrow syntax for object methods